api/server/middleware/user_agent.go

```diff
@@ -16,6 +17,7 @@ func NewUserAgentMiddleware(versionCheck string) Middleware {
 
 	return func(handler httputils.APIFunc) httputils.APIFunc {
 		return func(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
+			ctx = dockerversion.WithUserAgent(ctx, r.Header.Get("User-Agent"))
```

Comment:
you don't need a new function and importing the package here, just call:

```go
ctx = ctx.WithValue(ctx, theUserAgentConstantName, r.Header.Get("User-Agent"))
```

This might not be the best solution, but we do NOT want to import the dockerversion package here.